components/core/src/clp/BoundedReader.cpp

```diff
@@ -0,0 +1,39 @@
+#include "BoundedReader.hpp"
+
+namespace clp {
+auto BoundedReader::try_seek_from_begin(size_t pos) -> ErrorCode {
+    auto next_pos = pos > m_bound ? m_bound : pos;
```

Comment:
nit: if I am not mistaken, this can be a const
```suggestion
    auto const next_pos = pos > m_bound ? m_bound : pos;
```